cli.py

```diff
@@ -123,19 +123,29 @@
                 headers = {"Authorization": f"token {github_access_token}"}
                 response = requests.get("https://api.github.com/user", headers=headers)
                 if response.status_code == 200:
-                    scopes = response.headers.get("X-OAuth-Scopes")
+                    o_auth_scopes_header = "X-OAuth-Scopes"
+                    scopes = response.headers.get(o_auth_scopes_header, [])
                     if "public_repo" in scopes or "repo" in scopes:
                         click.echo(
                             click.style(
                                 "✅ GitHub access token has the required permissions.",
                                 fg="green",
                             )
                         )
+                    elif o_auth_scopes_header not in response.headers:
+                        install_error = True
+                        click.echo(
+                            click.style(
+                                "❌ Response is lack of X-OAuth-Scopes, please make sure you use OAuth token.",
+                                fg="red",
+                            )
+                        )
                     else:
                         install_error = True
                         click.echo(
                             click.style(
-                                "❌ GitHub access token does not have the required permissions. Please ensure it has 'public_repo' or 'repo' scope.",
+                                "❌ GitHub access token does not have the required permissions. "
+                                "Please ensure it has 'public_repo' or 'repo' scope.",
```

Comment:
**Suggestion:** The error message for missing permissions is split into two lines, which could be combined into one for better code readability and maintainability. [maintainability]
```suggestion
                                "❌ GitHub access token does not have the required permissions. Please ensure it has 'public_repo' or 'repo' scope."
```